source/app/controllers/sessions_controller.rb

```diff
@@ -4,9 +4,28 @@ def index
   end
 
   def create
+    user = User.find_by(email: session_params[:email]).try(:authenticate, session_params[:password])
+
+    if user
+      session[:user_id] = user.id
+      redirect_to user
+    else
+      flash.now[:alert] = "Email or password incorrect!"
+      render 'new'
+    end
   end
 
   def destroy
+    if session[:user_id]
```

Comment:
if this inherits from ApplicationController, its methods should be accessible here.